test/notifications/display_test.dart

```diff
@@ -166,30 +197,79 @@ void main() {
+        expectedTitle: expectedTitle,
         expectedTagComponent: expectedTagComponent);
+      testBinding.androidNotificationHost.clearActiveNotifications();
 
       testBinding.firebaseMessaging.onBackgroundMessage.add(
         RemoteMessage(data: data.toJson()));
       async.flushMicrotasks();
-      checkNotification(data, expectedTitle: expectedTitle,
+      checkNotification(data,
+        messageStyleMessages: [data],
+        expectedIsGroupConversation: expectedIsGroupConversation,
+        expectedTitle: expectedTitle,
         expectedTagComponent: expectedTagComponent);
+      testBinding.androidNotificationHost.clearActiveNotifications();
+    }
+
+    Future<void> receiveFcmMessage(FakeAsync async, MessageFcmMessage data) async {
+      testBinding.firebaseMessaging.onMessage.add(
+        RemoteMessage(data: data.toJson()));
+      async.flushMicrotasks();
     }
 
     test('stream message', () => awaitFakeAsync((async) async {
       await init();
       final stream = eg.stream();
       final message = eg.streamMessage(stream: stream);
       await checkNotifications(async, messageFcmMessage(message, streamName: stream.name),
+        expectedIsGroupConversation: true,
         expectedTitle: '#${stream.name} > ${message.topic}',
         expectedTagComponent: 'stream:${message.streamId}:${message.topic}');
     }));
 
+    test('multiple stream messages, same topic', () => awaitFakeAsync((async) async {
+      await init();
+      final stream = eg.stream();
+      final message1 = eg.streamMessage(topic: 'topic 1', stream: stream, timestamp: 1);
+      final data1 = messageFcmMessage(message1, streamName: stream.name);
+      final message2 = eg.streamMessage(topic: 'topic 1', stream: stream, timestamp: 2);
+      final data2 = messageFcmMessage(message2, streamName: stream.name);
+      final message3 = eg.streamMessage(topic: 'topic 1', stream: stream, timestamp: 3);
+      final data3 = messageFcmMessage(message3, streamName: stream.name);
+
+      await receiveFcmMessage(async, data1);
+      checkNotification(data1,
+        messageStyleMessages: [data1],
+        expectedIsGroupConversation: true,
+        expectedTitle: '#${stream.name} > ${message1.topic}',
+        expectedTagComponent: 'stream:${message1.streamId}:${message1.topic}');
+
+      await receiveFcmMessage(async, data2);
+      checkNotification(data2,
+        messageStyleMessages: [data1, data2],
+        expectedIsGroupConversation: true,
+        expectedTitle: '#${stream.name} > ${message2.topic}',
+        expectedTagComponent: 'stream:${message2.streamId}:${message2.topic}');
+
+      await receiveFcmMessage(async, data3);
+      checkNotification(data3,
+        messageStyleMessages: [data1, data2, data3],
+        expectedIsGroupConversation: true,
+        expectedTitle: '#${stream.name} > ${message3.topic}',
+        expectedTagComponent: 'stream:${message3.streamId}:${message3.topic}');
```

Comment:
Similar to https://github.com/zulip/zulip-flutter/pull/718#discussion_r1673223547 : this test is relying on all three of these messages being in the same stream and topic. (If they weren't, then the `messageStyleMessages: [data1, data2, data3]` above wouldn't be the right expectation.) So let's make that explicit in the structure of the test:
```suggestion
        expectedTitle: '#${stream.name} > $topic',
        expectedTagComponent: 'stream:${stream.streamId}:$topic');
```

by expressing the topic and stream ID the same way each time.

Or probably better yet: pull out locals for `expectedTitle` and `expectedTagComponent` and compute them just once.